modules/dkan/dkan_datastore/src/DatastoreInterface.php

```diff
@@ -0,0 +1,53 @@
+<?php
+namespace Dkan\Datastore;
+
+interface DatastoreInterface {
+
+  public function __construct($uuid);
+
+  /**
+   * Return the node (resource) associated with this datastore.
+   */
+  public function getNode();
+
+  /**
+   * A machine readable identifier for the datastore.
+   */
+  public function getId();
+
+  /**
+   * A label is a human readable identifier for the datastore.
+   */
+  public function getLabel();
+
+  /**
+   * Progress of the import (0 - 1).
+   */
+  public function getImportProgress();
+
+  /**
+   * A string with datastore status information.
+   */
+  public function getStatusMessage();
```

Comment:
I think some other getStatus() interface + methods should exist with predefined vars/structure, maybe start with Feed's fields (progress_importing, progress_clearing, imported, count, next) but more generically named so we can augment in the future for other engines

Reply:
I also like this idea, I will add this to a @todo. This abstraction might be necessary anyways once we start tackling queued imports (I am not doing that yet :) )